storage/database/group_storage.go

```diff
@@ -16,13 +18,33 @@ func NewGroupStorage(DB *Database) storage.IGroupStorage {
 }
 
 func (g *GroupStorage) AddGroup(group types.Group) error {
-	//TODO implement me
-	panic("implement me")
+	groupItem := MakeGroup(group)
+
+	// check if group already exists
+	_, err := g.GetGroupByID(groupItem.ID)
```

Comment:
For me it seems that we don't need this check. According to the docs (https://gorm.io/docs/advanced_query.html -> FirstOrCreate section) it is enough to only call FirstOrCreate and check the value of RowsAffected.

If RowsAffected == 0, it means that there already exists an user with given ID and we can return the "GroupAlreadyExistsError" in this case.

Reply:
Its changed.